XUtils.c

```diff
@@ -177,6 +178,8 @@ char* String_getToken(const char* line, const unsigned short int numMatch) {
 
       if (inWord == 1) {
          if (count == numMatch && line[i] != ' ' && line[i] != '\0' && line[i] != '\n' && line[i] != (char)EOF) {
+            if (foundCount == sizeof(match) / sizeof(match[0]) - 1)
```

Comment:
Better rewrite this function to copy the token in one go and allocate a properly sized target buffer instead of this incremental copy.

Reply:
I have removed the function after realizing that it is not used anymore.